stress/basic.go

```diff
@@ -0,0 +1,539 @@
+package stress
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/influxdb/influxdb/client/v2"
+)
+
+// AbstractTag is a struct that contains data
+// about a tag for in a series
+type AbstractTag struct {
+	Key   string `toml:"key"`
+	Value string `toml:"value"`
+}
+
+// AbstractTags is an slice of abstract tags
+type AbstractTags []AbstractTag
+
+func (a AbstractTags) Tag(i int) Tags {
+	tags := make(Tags, len(a))
+
+	for j, t := range a {
+		tags[j] = Tag{
+			Key:   t.Key,
+			Value: fmt.Sprintf("%v-%v", t.Value, i),
+		}
+	}
+
+	return tags
+}
+
+// Template returns a templated string of tags
+func (t AbstractTags) Template() string {
+	var buf bytes.Buffer
+	for i, tag := range t {
+		if i == 0 {
```

Comment:
Its building up the template string for the tagset in line protocol format.

so for if you have the tags 

```
{key: "host", value:"server"}
```
```
{key: "location", value: "us-west"}
```

Template will return `host=server-%v,location=us-west`